src/lib.rs

```diff
@@ -722,13 +674,11 @@ fn split_and_decode(s: &str) -> Result<(String, Vec<u5>), Error> {
             if c.is_lowercase() {
                 match case {
                     Case::Upper => return Err(Error::MixedCase),
-                    Case::None => case = Case::Lower,
```

Comment:
In e0bfa91dc74888005246ae80b46d0a620827888f:

Removing these `Case::None` cases introduces a bug in case the HRP is all digits but the data part has uppercase letters.

Reply:
Added this test, does this cover what you meant?
```rust
    
    #[test]
    #[cfg(feature = "alloc")]
    fn attempt_decode_all_digit_hrp_mixed_case_data() {
        let addr = "23451qar0sRRR7xfkvy5l643lydnw9re59gtzzwf5mdq";
        assert!(crate::decode(addr).is_err());
    }
```

Reply:
I believe this is ok now, I added a test to verify it.
```rust
    #[test]
    #[cfg(feature = "alloc")]
    fn decode_all_digit_hrp_uppercase_data() {
        let addr = "23451QAR0SRRR7XFKVY5L643LYDNW9RE59GTZZLKULZK";
        let (hrp, data, variant) = crate::decode(&addr).expect("address is well formed");
        assert_eq!(hrp, Hrp::parse("2345").unwrap().0);
        let (hrp, _) = Hrp::parse("2345").unwrap();
        let s = crate::encode(hrp, &data, variant).expect("failed to encode");
        assert_eq!(s, addr);
    }
```

Reply:
I added `Case::None` back in and added this test
```rust  
    #[test]
    #[cfg(feature = "alloc")]
    fn decode_all_digit_hrp_uppercase_data() {
        let addr = "23451QAR0SRRR7XFKVY5L643LYDNW9RE59GTZZLKULZK";
        let (hrp, data, variant) = crate::decode(&addr).expect("address is well formed");
        assert_eq!(hrp, Hrp::parse("2345").unwrap());
        let hrp = Hrp::parse("2345").unwrap();
        let s = crate::encode(hrp, &data, variant).expect("failed to encode");
        assert_eq!(s.to_uppercase(), addr);
    }
```